packages/data-broker/src/services/dhis/pullers/EventsPuller.js

```diff
@@ -28,17 +29,56 @@ export class EventsPuller {
       endDate,
       dataElementIdScheme: 'code',
     });
-
-    const translatedEventAnalytics = await this.translator.translateInboundEventAnalytics(
-      eventAnalytics,
+    const orgUnitEventAnalytics = await this.translator.translateInboundEventAnalytics(
+      orgUnitResults,
       dataElementSources,
     );
 
-    return buildEventsFromDhisEventAnalytics(
-      this.models,
-      translatedEventAnalytics,
-      dataElementCodes,
-    );
+    return buildEventsFromDhisEventAnalytics(this.models, orgUnitEventAnalytics, dataElementCodes);
+  };
+
+  pullEventsForApi = async (api, programCode, options) => {
+    const { organisationUnitCodes: entityCodes, hierarchy } = options;
+
+    if (!entityCodes || entityCodes.length === 0) {
```

Comment:
Minor: move validation to pull()? And should we add validation and a test for hierarchy present?

Reply:
Yes and yes 👍

Reply:
Okay, I double checked it this and it turns out that the deprecated event pull api in DHIS does support no org unit codes. Since the `DeprecatedEventsPuller` extends this class, it causes issues if I move the validation to `pull`, so keeping it here.